### PR TITLE
Bookstore: Use Redux in React components.

### PR DESCRIPTION
**Pull request by fahimy143**
### Project requirements 

- First of all - make sure that you keep state immutable in reducers!
- Secondly, change the default state in your books reducer from an empty array to an array with a few books. One of the goals is to display them in your React components.
- Import Redux Provider and your store in the main component of your React app.
- Connect your app to the Redux store.
- Make sure that list of books displays books from Redux store.
   - Use useSelector to consume the state.
- In your React component responsible for adding new books, add functionality for "Add new" button click:
- Import Redux useDispatch.
  - Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step).
  - Takes the inputs from the state and generate a unique id and add them to an object.
  - Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for removing books, add a functionality for "Remove" button click:
  - Import Redux useDispatch.
  - Dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for categories, add a functionality for "Check status" button click:
  - Import Redux useDispatch.
  - Dispatch a corresponding action (import Action Creator that you created before and use it here).

### CODE REVIEW

File: src/components/Form.js [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
+
+const Form = () => {
+  const dispatch = useDispatch();
+  const title = useRef(null);
+  const author = useRef(null);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(addBook(title.current.value, author.current.value, uniqid()));
+    e.target.reset();
```
Comment (codepantha) [inline src/components/Form.js]: - [x] You have demonstrated an understanding of the `useRef` hook which is good 👏 However, for the purpose of this milestone, you are required to have a local state with `title` and `author` and they should have their inputs that updates the local state every time they update (`onChange`)

File: src/components/Book.js [hunk 2]
```diff
@@ -1,36 +1,27 @@
...
-    </div>
+import { useSelector, useDispatch } from 'react-redux';
+import { removeBook } from '../redux/books/books';
 
-    <div className="card">
+const Book = () => {
+  const books = useSelector((state) => state.booksReducer);
+  const dispatch = useDispatch();
+  return (books.map((book) => (
+    <div key={book.id} className="card">
```
Comment (codepantha) [inline src/components/Book.js:8]: - [ ] [OPTIONAL] Usually this code which gets all the books in the store and iterates through should be done in the `Books` component. Your singular `Book` component should simply render the individual books that were passed in as props to it from the `Books` component👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️ 

## Hi @fahimy143 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ No linter errors
✔️ Correct GitFlow

## Required Changes ♻️
_Check the comments under the review._

- [x] In your README, under the Live Demo section, you still have Math Magicians. Please rename that portion and since your app is currently not deployed, you may want to replace it with a **Coming Soon** notice.
image

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
## Hello @fahimy143 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**